l5kit/l5kit/data/map_api.py

```diff
@@ -134,6 +142,81 @@ def get_lane_coords(self, element_id: str) -> dict:
+        """
+        Interpolate points based on cumulative distances from the first one. Two modes are available:
+        INTER_METER: interpolate using step as a meter value over cumulative distances (variable len result)
+        INTER_ENSURE_LEN: interpolate using a variable step such that we always get step values
+        Args:
+            xyz (np.ndarray): XYZ coords
+            cum_dist (np.ndarray): distances from the first coords of xyz. Same length as xyz
+            step (float): param for the interpolation
+            method (InterpolationMethod): method to use to interpolate
+
+        Returns:
+            np.ndarray: the new interpolated coordinates
+        """
+        if method == InterpolationMethod.INTER_ENSURE_LEN:
+            step = int(step)
+            assert step > 1, "step must be at least 2 with INTER_ENSURE_LEN"
+            steps = np.linspace(cum_dist[0], cum_dist[-1], step)
+
+        elif method == InterpolationMethod.INTER_METER:
+            assert step > 0, "step must be greater than 0 with INTER_FIXED"
+            steps = np.arange(cum_dist[0], cum_dist[-1], step)
+        else:
+            raise NotImplementedError("interpolation method unknown")
+
+        xyz_inter = np.zeros((len(steps), 3), dtype=xyz.dtype)
+        xyz_inter[:, 0] = np.interp(steps, xp=cum_dist, fp=xyz[:, 0])
+        xyz_inter[:, 1] = np.interp(steps, xp=cum_dist, fp=xyz[:, 1])
+        xyz_inter[:, 2] = np.interp(steps, xp=cum_dist, fp=xyz[:, 2])
+        return xyz_inter
+
+    @lru_cache(maxsize=CACHE_SIZE)
+    def get_lane_as_interpolation(self, element_id: str, step: float, method: InterpolationMethod) -> dict:
+        """
+        Perform an interpolation of the left and right lanes and compute the midlane.
+        See interpolate for details about the different interpolation methods
+
+        Args:
+            element_id (str): lane id
+            step (float): step param for the method
+            method (InterpolationMethod): one of the accepted methods
+
+        Returns:
+            dict: same as `get_lane_coords` but overwrite xyz values for the lanes
+        """
+        lane_dict = self.get_lane_coords(element_id)
+        xyz_left = lane_dict["xyz_left"]
+        xyz_right = lane_dict["xyz_right"]
+
+        # cumulative distance from the first point, including the first itself as 0
+        distances_left = np.cumsum(np.linalg.norm(np.diff(xyz_left, axis=0), axis=-1))
+        distances_left = np.insert(distances_left, 0, 0)
+
+        distances_right = np.cumsum(np.linalg.norm(np.diff(xyz_right, axis=0), axis=-1))
+        distances_right = np.insert(distances_right, 0, 0)
+
+        lane_dict["xyz_left"] = self.interpolate(xyz_left, distances_left, step, method)
+        lane_dict["xyz_right"] = self.interpolate(xyz_right, distances_right, step, method)
+
+        # compute midlane using mean of fixed length lanes
+        if method != InterpolationMethod.INTER_ENSURE_LEN:
```

Comment:
Perhaps add a comment to why the extra computation makes more sense if not using INTER_ENSURE_LEN. I almost forgot... and we had discussed Wed morning. So perhaps is useful to include it here. If you agree, but happy either way.